cudax/include/cuda/experimental/__stf/internal/cuda_kernel_scope.cuh

```diff
@@ -116,6 +116,24 @@ public:
   // move-constructible
   cuda_kernel_scope(cuda_kernel_scope&&) = default;
 
+  /// Add a set of dependencies
+  template <typename... Pack>
+  void add_deps(task_dep_untyped first, Pack&&... pack)
+  {
+    dynamic_deps.push_back(mv(first));
+    if constexpr (sizeof...(Pack) > 0)
+    {
+      add_deps(::std::forward<Pack>(pack)...);
+    }
```

Comment:
Q: should the push only happen if there are actually dependencies, aka inside the `if constexpr`?